src/app/buildings/components/building.component.ts

```diff
@@ -0,0 +1,73 @@
+import { rxResource } from '@angular/core/rxjs-interop';
+import { ChangeDetectionStrategy, Component, computed, effect, inject, input, output, Signal } from '@angular/core';
+import { BuildingService } from '../services/building.service';
+import { DEFAULT_CENTER, DEFAULT_MAP_CONFIG } from '../../maps/utils/constants';
+import { MapMarker } from '../../maps/interfaces/map-marker.interface';
+import { MapConfig } from '../../maps/interfaces/map-config.interface';
+import { of } from 'rxjs';
+import { BuildingDetails } from '../interfaces/building-details.interface';
+
+
+@Component({
+  selector: 'app-building-component',
+  imports: [],
+  template: ``,
+  styleUrl: './building.component.css',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class BuildingComponent {
+  buildingService = inject(BuildingService);
+
+  buildingsResource = rxResource({
+    loader: () => {
+      return this.buildingService.getBuildings();
+    }
+  });
+
+  buildingMarkers: Signal<MapMarker[]> = computed(() => (this.buildingsResource.value() ?? []).map(building => ({
+    id: building.buildingId,
+    coordinate: {
+      latitude: building.latitude,
+      longitude: building.longitude,
+    },
+    icon: {
+      url: '/building.png'
+    }
+  })));
+
+  private mapConfig: Signal<MapConfig> = computed(() => ({
+    center: DEFAULT_CENTER,
+    zoom: 14.5,
+    enableClick: true,
+    enableControls: false,
+    markers: this.buildingMarkers()
+  }));
```

Comment:
En este caso, los atributos que no son marcadores son reemplazados automáticamente por la configuración por defecto del componente de mapas.

La idea de las señales computadas es que reflejen exactamente qué es lo que necesita el usuario. Refactoricemos.

```suggestion
  private mapConfig: Signal<MapConfig> = computed(() => ({
    markers: this.buildingMarkers()
  }));
```